playwright/tests/e2e/steps/transactions-overview.steps.ts

```diff
@@ -0,0 +1,221 @@
+  const transactionDetailFromWidgetDetailViewBtn = page.getByTestId("transaction.detailViewEpoch.viewDetail");
+  transactionHash = await transactionDetailFromWidgetDetailViewBtn.getAttribute("href");
+  await transactionDetailFromWidgetDetailViewBtn.click();
+});
+Then(/^the transaction details page of the selected transaction should be opened$/, async ({ page }) => {
+  await transactionDetailPage(page).checkTransactionsDetail({ transactionHash: transactionHash || "" });
+});
+
+Given(/^the user is in the Transactions page to address detail from detail button in widget$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToTransactions();
+});
+Given(/^the user open the info widget of one transaction record in transaction web$/, async ({ page }) => {
+  await transactionOverviewPage(page).openLastestTransactionWidget();
+});
+When(/^the user selects the Input or Output address hash in the transcation info widget$/, async ({ page }) => {
+  const addressDetailProducer = page.getByTestId("transaction.widget.inputAddress");
+  addressId = await addressDetailProducer.getAttribute("href");
+  await addressDetailProducer.click();
+});
+Then(
+  /^the user should be redirected to the address detail page of the selected input address hash$/,
+  async ({ page }) => {
+    await transactionDetailPage(page).checkAddressDetail({ addressId: addressId || "" });
+  }
+);
+
+Given(
+  /^the user is in the Transactions page for go to transaction detail from summary in widget$/,
+  async ({ page }) => {
+    await transactionOverviewPage(page).goToTransactions();
+  }
+);
+Given(
+  /^the user opens the info widget of any transaction in the transactions table from widget of transaction$/,
+  async ({ page }) => {
+    await transactionOverviewPage(page).openLastestTransactionWidget();
+  }
+);
+When(/^the user selects the summary section in the info widget$/, async ({ page }) => {
+  const transactionDetailFromWidgetBySumary = page.getByTestId("transaction.widget.summary");
+  transactionHash = await transactionDetailFromWidgetBySumary.getAttribute("href");
+  await transactionDetailFromWidgetBySumary.click();
+});
+Then(
+  /^the transaction detail page of the selected transaction should be opened with the summary section displayed$/,
+  async ({ page }) => {
+    await transactionDetailPage(page).checkTransactionsDetail({ transactionHash: transactionHash || "" });
+  }
+);
+
+Given(/^the user is in the Transactions page for go to transaction detail from widget$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToTransactions();
+});
+Given(/^the user opens the info widget of any transaction in the transactions table from widget$/, async ({ page }) => {
+  await transactionOverviewPage(page).openLastestTransactionWidget();
+});
+When(/^the user selects the UTXOs section in the info widget of the selected transaction$/, async ({ page }) => {
+  const transactionDetailFromWidgetByUTXO = page.getByTestId("transaction.widget.utxOs");
```

Comment:
let's keep locators interactions inside the pages functions and returns the values that you could need in the next steps if need it.